File: Swift-C++-IR-Comparison/test/functions/Swift/FunctionOptionalReturn.swift
@main
struct MyMain {

    static func main() {
        let myNil = identity(x: 42)
        let myIdentity = identity(x: 0)
    }
    static func identity(x: Int) -> Int? {
        if x == 0 {
            return x
        }
        return nil
    }
}

File: Swift-C++-Testsuite/AutoTranspiledTests/try-0/FunctionOptionalReturn.cpp
 
#include <iostream>

using namespace std;

struct MyMain
{
	static int identity(int x)
	{
		if (x == 0)
		{
			return x;
		}
		return NULL;
	}
	
};

int main()
{
	int myNil = MyMain::identity(42); 
	int myIdentity = MyMain::identity(0);
	return 0;
}